packages/embed/src/utils/build-iframe-src.ts

```diff
@@ -29,7 +31,8 @@ export const buildIframeSrc = (params: BuildIframeSrcOptions): string => {
   }
 
   const queryParams = mapOptionsToQueryParams(type, options)
-  Object.entries(queryParams).forEach(([paramName, paramValue]) => {
+  const queryWithTransferableParams = getTransferableQueryParams(queryParams, options.transferableUrlParameters)
```

Comment:
I think this belongs to the `mapOptionsToQueryParams`, since we already pass `options.transferableUrlParameters` to it. I would like to keep all this query building logic in one place.

Also, we could simplify `getTransferableQueryParams` to not check if the param already exists in `queryParams`, we can just override those params:

```
const mapOptionsToQueryParams = (type: EmbedType, options: UrlOptions): Record<string, any> => {
  const transferredParams = getTransferableQueryParams(options.transferableUrlParameters)
  const params = {
    'typeform-embed': typesToEmbed[type],
    'typeform-source': options.source,
    // etc...
  }
  return { ...transferredParams, ...params }
}
```

Reply:
yep, makes sense